test/e2e/fleet_test.go

```diff
@@ -602,6 +603,49 @@ func TestGameServerAllocationDuringGameServerDeletion(t *testing.T) {
 	})
 }
 
```

Comment:
Do you think it makes sense to edit `scaleFleet` to use the new UpdateScale across all the e2e tests?

If we do this, we don't need the extra new test below, it's tested across the board.

WDYT?

Reply:
I can leave both old function with scaleFleet and new method with UpdateScale() so there would be two iterations in such test functions.

Reply:
SGTM! Test both!